### DISCUSSION TITLE
Maximum compression of builds

### DISCUSSION THREAD

**Discussion by rasa** [okibcn/nano-for-windows#19] [Show and tell]
Per https://superuser.com/a/742034, I created [this branch](https://github.com/rasa/nano-for-windows/tree/rasa-max-compression) and got [this](https://github.com/rasa/nano-for-windows/actions/runs/4208035110). Here are the results:

| Branch | 7z | zip |
| - | -: | -: |
| [releases](https://github.com/okibcn/nano-for-windows/releases/tag/v7.2-13-10168) | 611,650 | 757,442 |
| [rasa-max-compression](https://github.com/rasa/nano-for-windows/suites/11057055872/artifacts/562072062) | 612,231 | 731,935 |
| bytes saved | -581<br/>(-0.09%) | 25,507<br/>(3.37%) | 

7z compression is tricky, so maybe we just stick with the defaults. Is the zip compression worth a PR?

**Comment by okibcn** [comment]
The package is so small that the DL size is not a problem. Compatibility with default windows is more important here, and sadly, zip is available out of the box in Windows. those opts for ZIP are good.

At this time it is not an issue, but in the future, I would like to deliver 6 files per release:
Nano-For-Windows_vN.N-N.N-{win32,win64,arm64}.{zip,exe}

That would reduce a lot the DL size, about half.

Another possible improvement if all the binaries are delivered in the same package is to create a unique syntax and docs folder in the delivery structure. I wanted to maintain the linux-like folder structure used by lhmouse in his releases, but that wasn't my preference. I believe more package size can be saved that way than the small gain using 7z exe

And I really prefer 7z over zip. Even gz or xz are better than zip.

**Comment by okibcn** [comment]
I've done some tests:
7z format:
611958 bytes (7zz binary, the original)
611755 bytes (7z binary, the independent implementation)
611494 bytes (7zz binary) with `-mx9` option.
611282 bytes (7z binary) with `-mx9` option.

ZIP format.
753745 (7zz binary)
753745  (7z binary)
750058 (7zz binary) with `-mm=Deflate -mfb=258 -mpass=15`  options.
750058 (7z binary) with `-mm=Deflate -mfb=258 -mpass=15`  options.


Note I just used your zipped file, extracted the files, and compressed them with these options.
